src/common/lambda.rs

```diff
@@ -54,12 +55,12 @@ impl Lambda {
     }
 
     /// Emits an opcode as a byte.
-    pub fn emit(&mut self, op: Opcode) {
+    pub unsafe fn emit(&mut self, op: Opcode) {
```

Comment:
I totally agree, I think implementing the verification step you mentioned to make the code generation safe would be a good solution here! I was deliberately avoiding making any actual code changes in this PR so that we could decide how we'd refactor the unsafety out.

Reply:
Unsafe because of bounds checking (when reading the data before they are emitted)?

Reply:
Here there is no bounds checking - we're converting an enum variant to a byte. The variant is always within bounds because of Rust's type system, so this function _alone_ is not unsafe. The closest comparison I could think of would be, if you have code that writes an executable, is the code that outputs binary to the executable file itself unsafe? I would say no, because writing to a file is well defined, even if we end up with a malformed executable. It's the act of executing the executable in an unsandboxed environment that's unsafe, if that makes sense. ¯\_(ツ)_/¯

Reply:
Yeah, that makes sense. Thanks for clarification!